### ISSUE TITLE
[Feature Request] Ability to unsubscribe from events

### ISSUE THREAD

**Issue by jessethomson** [ShafSpecs/near-orm#1]
First off, great work on this library. Very impressive. I know this is very new so I don't know if you are open to other people opening issues or contributing, so feel free to close any issues I open if you don't have the time to be able to address them. With that said, I see a lot of potential with this library and would love to provide feedback and whatever else I can to help.

# Feature Request

With `near-orm` I can subscribe to events like so:
```
db.events.on('create', (storeName, data) => {
  console.log(`New record created in ${storeName}:`, data);
});
```

However, I would also like to be able to _unsubscribe_ from those events later. This would be great to avoid memory leaks for event handlers that are no longer needed. It would also make it easier for frameworks to create abstractions for listening to changes.

For example, in React, you could do:
```
useEffect(() => {
    const unsubscribe = db.events.on('create', (storeName, data) => {
        // some subscription logic
    });

    return () => unsubscribe();
}, []);
```

# Proposed Solution

I would propose that calling `db.events.on` should return an unsubscribe function that allows you to unsubscribe that handler from the event.

# Implementation

This could be achieved pretty easily. 

At the end of this function:
https://github.com/ShafSpecs/near-orm/blob/088b2d4ac2a608d98d8b58761a5f8e3a33f254d3/orm.ts#L455-L463

You can add something like:
```
const unsubscribe = () => {
	const index = this.eventHandlers[eventName].indexOf(callback);

	if (index !== -1) {
		this.eventHandlers[eventName].splice(index, 1);
	}
};

return unsubscribe;
```

**Comment by ShafSpecs** [comment]
First off, thanks for the suggestion. Contributions and issues are very welcome, also, your request is a good one ❤️.

I think this even opens up the potential for a `once` api as well, to listen to an even once and discard the callback immediately after.

**Comment by ShafSpecs** [comment]
Released [v0.3.0](https://www.npmjs.com/package/near-orm/v/0.3.0) which closes this issue.

Thanks for your contribution 🫡!

**Comment by jessethomson** [comment]
Great, appreciate it!
